### DISCUSSION TITLE
Proper way to display volume

### DISCUSSION THREAD

**Discussion by PhlegmaTREEc** [qtile/qtile#3538] [Q&A] [answered]
Hello,
I've been trying to work out how to best display current volume level on my panel. What I've tried so far:
1) `widget.PulseVolume(),`
kinda works, but when I switch audio source, I have to reload Qtile for the widget to use the the new source
2) `widget.GenPollText(update_interval=1, func=lambda: subprocess.check_output("/home/ptc/.config/qtile/scripts/volume.sh")),`
script is simply `pamixer --get-volume-human`
This works and also auto updates the source, but displays extra symbols in the widget - b'50%\n'
50% is correct, but that extra stuff not so much.
Any ideas?

**Comment by elParaguayo** [comment, marked as answer]
You could edit your `volume.sh` script to tidy up the output before returning it.

Or you could try the `Volume` widget and set the `get_volume_command` but I think this expects the volume in the output to be in square brackets.

**Reply by PhlegmaTREEc** [reply]
I would appreciate tips on how tidy up the script. In terminal the script returns just what I want - 50%.

**Reply by elParaguayo** [reply]
Sadly, I'm no good with bash scripts at all.

I wonder if we should just let `GenPollText` have a `parse` attribute where we can provide a python function to parse the text. `GenPollUrl` has this so no reason not to let `GenPollText` have one too.

**Reply by hastiin-manuelito** [reply]
In bash, you might use printf to format the output; it looks like qtile is reading the output in bytes (the usage of  `b'`). An alternative could be to use a python script instead, using the awesome formatting capabilities. I am pretty sure that will play much better with qtile.

**Reply by elParaguayo** [reply]
@hastiin-manuelito exactly. If we add a `parse` attribute then we can create python functions to manipulate the output of any script.

**Reply by elParaguayo** [reply]
Sorry, I've been really dumb. Give me a few mins and I'll post something.

**Reply by elParaguayo** [reply]
Just try changing your lambda function to `subprocess.check_output("/home/ptc/.config/qtile/scripts/volume.sh").decode().strip()`

**Reply by PhlegmaTREEc** [reply]
> Just try changing your lambda function to `subprocess.check_output("/home/ptc/.config/qtile/scripts/volume.sh").decode().strip()`

Works like a charm. Thank you
